### ISSUE TITLE
Why Code 8 using MSE loss other than inner product of sphere?

### ISSUE THREAD

**Issue by Luciennnnnnn** [facebookresearch/flow_matching#30]
Hi, according to section 5.2, I expect that using Loss function: D_x(u, v) = ||u − v||_g^2, where distance is defined through Riemannian inner product of sphere.

But Code 8 uses MSE Loss function without explanation, why do not use `loss = manifold(projected_model_output, sample.dx_t)`

**Comment by Luciennnnnnn** [comment]
I notice from example, there is a comments:
```python
# velocity field model init
vf = ProjectToTangent(  # Ensures we can just use Euclidean divergence.
    MLP(  # Vector field in the ambient space.
        input_dim=dim,
        hidden_dim=hidden_dim,
    ),
    manifold=manifold,
)
```

Why by projecting to tangent space, then we can just use euclidean divergence?

**Comment by Luciennnnnnn** [comment]
I see from appendix J.2 Riemannian norm of the paper "Flow Matching on General Geometries", using Riemannian norm should by important.

**Comment by rtqichen** [comment]
For the manifolds we released in this repo (flat tori and hypersphere), the Riemannian norm is equivalent to ||.||_2 on the tangent space, so for simplicity, we just used MSELoss to be similar to the Euclidean example. If you are interested, more manifolds are supported in https://github.com/facebookresearch/riemannian-fm where the code uses the [Riemannian metric during loss computation](https://github.com/facebookresearch/riemannian-fm/blob/main/manifm/model_pl.py#L615) --- though it is equivalent to MSE on the flat tori and hypersphere, manifolds implemented here.

> Why by projecting to tangent space, then we can just use euclidean divergence?

See Eq (32) of "Flow Matching on General Geometries".

> I see from appendix J.2 Riemannian norm of the paper "Flow Matching on General Geometries", using Riemannian norm should by important.

Yes for general manifolds with a metric tensor, we would need a different formula. See Eq (33). This is supported in the full [riemannian-fm repo](https://github.com/facebookresearch/riemannian-fm/blob/main/manifm/model_pl.py#L475-L481).
